public/app/features/apiserver/client.ts

```diff
@@ -90,3 +97,13 @@ export class ScopedResourceClient<T = object, K = string> implements ResourceCli
       .join(',');
   }
 }
+
+// add the origin annotations so we know what was set from the UI
+function setOriginAsUI(meta: Partial<ObjectMeta>) {
+  if (!meta.annotations) {
+    meta.annotations = {};
+  }
+  meta.annotations.AnnoKeyOriginName = 'UI';
+  meta.annotations.AnnoKeyOriginPath = window.location.pathname;
+  meta.annotations.AnnoKeyOriginHash = config.buildInfo.versionString;
```

Comment:
based on [this](https://github.com/grafana/grafana/blob/399651b9ad822dc0569fbf6efcbaebce6b03bdae/pkg/apimachinery/utils/meta.go#L41), isn't `AnnoKeyOriginHash` used for the dashboard checksum?

Reply:
The meaning and usage of hash will depend on the origin.  But ya weird.

For things from our existing provisioning system, the hash is the file checksum

Reply:
Alternative would be appending the version to path -- I don't have a strong sense of which approach is better

Reply:
(name should stay low cardinality and eventually link to origins with that name!)

Reply:
> meaning and usage of hash will depend on the origin

this makes sense to me as long as we'll never need version info at the same time as other uses of the hash